taichi/codegen/codegen_llvm.cpp

```diff
@@ -72,16 +72,20 @@ FunctionCreationGuard::FunctionCreationGuard(
 }
 
 FunctionCreationGuard::~FunctionCreationGuard() {
-  mb->builder->CreateRetVoid();
+  if (!mb->returned) {
```

Comment:
Does `finalize_offloaded_task_function()` already ensure the existence of return?

Reply:
They are used in different places. Take range-for for example, `initialize_offloaded_task_function` and `finalize_offloaded_task_function` creates and finalizes a offloaded task function. In the offloaded task function, the codegen uses `FunctionCreationGuard` to create a function that contains the body of the for loop (let's call it the body function). Then, the offloaded task function calls `cpu_parallel_range_for` to execute the body function in parallel.